src/Thentos/Backend/Core.hs

```diff
@@ -101,6 +101,8 @@ actionErrorToServantErr e = do
         { errBody = "error accessing user info" }
     _thentos (SsoErrorCouldNotGetAccessToken _) = pure $ err500
         { errBody = "error retrieving access token" }
+    _thentos (A3BackendError msg) = pure $ err500
+        { errBody = "error talking to A3 backend: " <> cs msg }
```

Comment:
I just pushed 6533ee7 regarding this.

A3BackendError is very untyped.  There is no strong reason to not do that besides that it's not necessary and not idiomatic.  I would use different types `A3BackendBadResponseCode`, `A3BackendNoSuchUser`, ..., and make each carry the details in typed form.  See 6533ee7 for more explanation.